scripts/copy_notification_image.sh

```diff
@@ -59,23 +52,32 @@ fi
 FULL_FILENAME="${IMG_PREFIX}${FULL_FILENAME}"
 mv -f "${IMAGE_TO_USE}" "${ALLSKY_HOME}/${FULL_FILENAME}"	# so web severs can see it.
 
-# If upload is true, optionally create a smaller version of the image and upload it
+# If upload is true, optionally create a smaller version of the image, either way, upload it.
 if [ "${UPLOAD_IMG}" = "true" ] ; then
-	# Don't overwrite FULL_FILENAME since the web server(s) may be looking at it.
-	cp "${ALLSKY_HOME}/${FULL_FILENAME}" "${ALLSKY_TMP}"
-	IMAGE_TO_USE="${ALLSKY_TMP}/${FULL_FILENAME}"
+	IMAGE_TO_USE="${ALLSKY_HOME}/${FULL_FILENAME}"
+
 	if [ "${RESIZE_UPLOADS}" = "true" ]; then
+		# Don't overwrite IMAGE_TO_USE since the web server(s) may be looking at it.
+		cp "${IMAGE_TO_USE}" "${ALLSKY_TMP}/${FULL_FILENAME}"  # create temporary copy to resize
+		IMAGE_TO_USE="${ALLSKY_TMP}/${FULL_FILENAME}"
 		# Create a smaller version for upload
 		convert "${IMAGE_TO_USE}" -resize "${RESIZE_UPLOADS_SIZE}" -gravity East -chop 2x0 "${IMAGE_TO_USE}"
 		if [ $? -ne 0 ] ; then
 			echo "${RED}*** ${ME}: ERROR: RESIZE_UPLOADS failed${NC}"
+			# Leave temporary file for possible debugging.
 			exit 4
-    		fi
+		fi
 	fi
 
 	# We're actually uploading $IMAGE_TO_USE, but show $NOTIFICATIONFILE in the message since it's more descriptive.
-	echo -e "${ME}: Uploading $(basename "${NOTIFICATIONFILE}")\n"
+	echo "${ME}: Uploading $(basename "${NOTIFICATIONFILE}")"
 	"${ALLSKY_SCRIPTS}/upload.sh" --silent "${IMAGE_TO_USE}" "${IMGDIR}" "${FULL_FILENAME}" "NotificationImage"
-	exit $?
+	RET=$?
+
+	# If we created a temporary copy, delete it.
+	[ "${RESIZE_UPLOADS}" = "true" ] && rm -f "${IMAGE_TO_USE}"
```

Comment:
I realize this contains `${ALLSKY_TMP}/${FULL_FILENAME}`, but let's use the expanded version ( as I provided it here ), instead of `${IMAGE_TO_USE}`.  This helps eliminate confusion during future code reviews.

Reply:
@linuxkidd, not sure I understand your comment.  Almost all the scripts use `${IMAGE_TO_USE}` as the name of the current image being processed.  This makes it easier to change locations of the file (like putting in in $ALLSKY_TMP in the near future) since only one line per script needs to change.

What do you mean by "as I provided it here"?  I only see a few lines of newly added code in your reply that removes a temporary file.